app/models/event.rb

```diff
@@ -1,6 +1,12 @@
 class Event < ApplicationRecord
-  has_many :tickets
   has_many :participants
+  has_many :tickets
 
   validates :name, presence: true
+  validates :quota,
+    presence: true,
+    numericality:
+    { greater_than_or_equal_to: 1,
+      less_than_or_equal_to: 1000,
+      allow_blank: true }
```

Comment:
Why allow blank ?

Reply:
As there is presence check before numericality check,  `allow blank` is set to avoid redundant messages in receiving blank input.

Reply:
Is that Rails behavior?

Reply:
I wonder presence and allow_blank.

Reply:
Which validation do you want ?

Reply:
Without allow_blank for numericality (not for presence), both presence and numericality validation set redundant error messages for one blank input as below:
- `quota can't be blank.`
- `quota must be a number.`

Of course, we can use only numericality validation, but I think `quota can't be blank.` message is more appropriate for blank input. So, I use both presence and numericality validation avoiding redundancy by setting `allow_blank` for numericality validation.

Reply:
Hmm..

Why do you wonder about it?

Reply:
it's correct that both messages are displayed when quota value is nil.

Reply:
Because numericality === is not nil.

Reply:
remove `allow_blank'

```
event = Event.new()
event.valid?
p event.errors.messages
#=>  name => ["を入力してください"],
     quota => ["を入力してください", "は数値で入力してください"]
```

Reply:
```
event = Event.new(quota: 0)
event.valid?
p event.errors.messages
#=> name => ["を入力してください"],
    quota => ["は1以上の値にしてください"]
```

Reply:
```
event = Event.new(quota: 'a')
event.valid?
p event.errors.messages
#=> name => ["を入力してください"],
    quota => ["は数値で入力してください"]
```

Reply:
for backend

Reply:
I see! 
If it is correct that both messages are displayed together,  I would remove `allow_blank`.